arithmetization/src/main/java/net/consensys/linea/zktracer/module/blake2fmodexpdata/Blake2fModexpData.java

```diff
@@ -40,35 +51,59 @@ public String moduleKey() {
 
   @Override
   public void enterTransaction() {
-    this.state.enter();
+    this.operations.enter();
+  }
+
+  @Override
+  public void traceEndTx(
+      WorldView worldView,
+      Transaction tx,
+      boolean isSuccessful,
+      Bytes output,
+      List<Log> logs,
+      long gasUsed) {
+    final List<Blake2fModexpDataOperation> newOperations =
+        new ArrayList<>(this.operations.sets.getLast())
+            .stream().sorted(Comparator.comparingLong(Blake2fModexpDataOperation::id)).toList();
+
+    this.sortedOperations.addAll(newOperations);
+    final int numberOfOperationsAtEndTx = sortedOperations.size();
+    for (int i = numberOfOperationsAtStartTx; i < numberOfOperationsAtEndTx; i++) {
+      final long previousID = i == 0 ? 0 : sortedOperations.get(i - 1).id();
+      this.wcp.callLT(previousID, sortedOperations.get(i).id());
+    }
+  }
+
+  @Override
+  public void traceStartTx(WorldView worldView, Transaction tx) {
+    this.numberOfOperationsAtStartTx = operations.size();
   }
 
   @Override
   public void popTransaction() {
-    this.state.pop();
+    this.sortedOperations.removeAll(this.operations.sets.getLast());
```

Comment:
Could you add a comment about what is the logic here?

Reply:
logic is the same as in shakira:

- during the tx we store the op in the stackedSet
- at the end of the tx, we 1) sort (by frowing id) the op of this tx
-  2) add this tx to sortedOperation

3) trigger wcp to prove the sorting



- if we pop a tx, we have to delete the op added by this tx, this is done by `
    this.sortedOperations.removeAll(this.operations.sets.getLast());`